src/Http/Http.Abstractions/src/IBindableFromHttpContextOfT.cs

```diff
@@ -0,0 +1,22 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Reflection;
+
+namespace Microsoft.AspNetCore.Http;
+
+/// <summary>
+/// Defines a mechanism for creating an instance of a type from an <see cref="HttpContext"/> when binding parameters for an endpoint
+/// route handler delegate.
+/// </summary>
+/// <typeparam name="TSelf">The type that implements this interface.</typeparam>
+public interface IBindableFromHttpContext<TSelf> where TSelf : class?, IBindableFromHttpContext<TSelf>
```

Comment:
```suggestion
public interface IBindableFromHttpContext<TSelf> where TSelf : class, IBindableFromHttpContext<TSelf>
```

Does this work though?

Reply:
Oh, hmm, I'll have to check again.

Reply:
Yeah seems to.

Reply:
So, which do we do? https://github.com/dotnet/aspnetcore/pull/41100#discussion_r848896183

Reply:
It looks like I left this comment a week later than the other one. It's also what's on the [approved API](https://github.com/dotnet/aspnetcore/issues/40927#issuecomment-1101622622). We covered my concerns about nullability by making the return type `ValueTask<TSelf?>` so we don't need the `?` on the `class`, so I'd go this suggestion if you don't see a problem with it.